Lib/ufo2ft/postProcessor.py

```diff
@@ -113,6 +113,10 @@ def process(
 
         if compileTrueTypeHinting and "glyf" in self.otf:
             self._compile_truetype_hinting()
+            # Force compilation of the font to avoid glyph name problems
+            tmp = BytesIO()
+            self.otf.save(tmp)
+            self.otf = TTFont(tmp)
```

Comment:
Here I had to resort to a kind of hack ... forcing the font to be compiled once by saving it and opening it again, otherwise the glyph names go out of sync and glyphs will disappear from the font.

Reply:
why do we need to compile the hinting in the postProcessor, can't we do in the outlineCompiler itself?

Reply:
I really don't like this recompilation step, we need to go to the bottom of the issue and try to avoid this.